pkg/upgrade/upgrade.go

```diff
@@ -308,6 +318,30 @@ func UpdateFromLegacyVersion(cli client.Client, platform deploy.Platform) error
 	return nil
 }
 
+func deleteDeployments(cli client.Client) error {
+	// In v2, Deployment selectors use a label "app.opendatahub.io/<componentName>" which is
+	// not present in v1. Since label selectors are immutable, we need to delete the existing
+	// deployments and recreated them.
+
+	// Delete Deployment objects
+	deployments := &appsv1.DeploymentList{}
+	listOpts := &client.ListOptions{
+		Namespace: "rhods-ods-applications",
+	}
+	err := cli.List(context.TODO(), deployments, listOpts)
+	if err != nil {
+		return err
+	} else {
+		for _, deployment := range deployments.Items {
+			err = cli.Delete(context.TODO(), &deployment, []client.DeleteOption{}...)
+			if err != nil {
+				return fmt.Errorf("error deleting deployment %v : %v", deployment.Name, err)
```

Comment:
should we use multiErr to catch all failed deletion? do not return on the first falied one